graviton/sim.py

```diff
@@ -0,0 +1,108 @@
+from dataclasses import dataclass
+from typing import Any, Dict, Iterable, List, Optional, Sequence, TypeVar, Union, cast
+
+from algosdk.v2client.algod import AlgodClient
+
+from graviton.abi_strategy import ABICallStrategy
+from graviton.blackbox import DryRunExecutor, DryRunTransactionParams as TxParams
+from graviton.inspector import DryRunProperty as DRProp, DryRunInspector
+from graviton.invariant import Invariant
+from graviton.models import ExecutionMode, PyTypes
+
+# TODO: this will encompass strategies, composed of
+# hypothesis strategies as well as home grown ABIStrategy sub-types
+InputStrategy = Union[Iterable[Sequence[PyTypes]], ABICallStrategy]
+
+
+@dataclass(frozen=True)
+class SimulationResults:
+    succeeded: bool
+    simulate_inspectors: List[DryRunInspector]
+    identities_inspectors: Optional[List[DryRunInspector]] = None
+
+
+class Simulation:
```

Comment:
I guess this class serves as the purpose of `ABIContractExecutor.dry_run_on_sequence`, together with some assertions in predicates?

Reply:
Yes, exactly. So basically:

`Simulation = (Teal logic for execution) + (predicates that must be satisfied)`

`run_and_assert: simulation + InputStrategy` &rarr; `SUCCESS or FAILURE`

In general, graviton needs a deep cleaning/update of it's documentation. I've just created #51 as a reminder

Reply:
sounds good, I was keeping track of deleted `ABIContractExecutor` and its methods, trying to find where are the new substitutions. So far I think these mechanisms have their replacements in place.